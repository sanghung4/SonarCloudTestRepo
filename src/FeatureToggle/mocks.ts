import { Feature } from 'generated/graphql';

export const mockFeatures: Feature[] = [
  { id: '1', isEnabled: false, name: 'noooo' },
  { id: '2', isEnabled: true, name: 'fine' },
  { id: '3', isEnabled: true, name: 'test' }
];
