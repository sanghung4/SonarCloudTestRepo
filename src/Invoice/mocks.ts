import { GetInvoiceDocument } from 'generated/graphql';

export const successMincron = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '123456',
      invoiceNumber: '5460026'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5460026',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: '40251/13325',
        invoiceDate: '10/28/2020',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'AVLNEP7',
        jobName: 'NEP 7 ASHEVILLE',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2020',
        deliveryMethod: 'Our Truck',
        subtotal: '3184.28',
        tax: '262.76',
        otherCharges: '0.0',
        paidToDate: '1000.00',
        invoiceItems: [
          {
            id: '1738433F4-91A7-4637-8C60-B4DF59A060C1',
            brand: 'Advanced Drainage Systems, Inc.',
            name: '120 x 7.5, 100 Sq Yd, Geosynthetic, Double, Photodegradable, Erosion Control Blanket',
            partNumber: '8X1125STRAWDN',
            mfr: '00S2TT',
            thumb: 'null',
            price: '43.59',
            qty: {
              quantityOrdered: 1,
              quantityShipped: 1
            }
          }
        ]
      }
    }
  }
};

export const successMincron_5448540 = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '91492',
      invoiceNumber: '5448540'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5282411',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: 'WILKESBORO JIM',
        invoiceDate: '10/06/2021',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'SHOP',
        jobName: 'SHOP',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2021',
        deliveryMethod: 'Our Truck',
        subtotal: '2184.28',
        tax: '162.73',
        otherCharges: '0.0',
        paidToDate: '139.91',
        invoiceItems: {
          id: 'F02FD85F-4DCE-4EA7-8459-2DA7CFD689DC',
          brand: 'Certex USA, name:2 x 12, Nylon, Eye and Eye, Ply Sling',
          partNumber: 'SLING212,mfr:CX07-0126-12',
          thumb:
            'https://images.tradeservice.com/ProductImages/DIR100193/CERTEX_CX07-0126-12_LINE_SML.jpg',
          price: '40.00',
          qty: {
            quantityOrdered: 1,
            quantityShipped: 1
          }
        }
      }
    }
  }
};
