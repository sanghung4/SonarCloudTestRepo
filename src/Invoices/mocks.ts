import { subMonths } from 'date-fns';
import { Invoice, InvoicesDocument } from 'generated/graphql';
import { mockRow } from 'test-utils/mockTableInstance';
import { formatDate } from 'utils/dates';

export const success = {
  request: {
    query: InvoicesDocument,
    variables: {
      accountId: '123456',
      erpName: 'ECLIPSE'
    }
  },
  result: {
    data: {
      invoices: {
        bucketNinety: 0.0,
        bucketOneTwenty: 0.0,
        bucketSixty: 0.0,
        bucketThirty: 0.0,
        bucketFuture: 0.0,
        currentAmt: 35224.96,
        invoices: [
          {
            age: 'Future',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227972.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: 'Current',
            customerPo: '1096872',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227975.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: 'Over120',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110220294.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: '91-120',
            customerPo: '109687',
            invoiceDate: '03/01/2021',
            dueDate: '03/01/2022',
            invoiceNumber: 'S110222345.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: 'Deposit',
            customerPo: '109687',
            invoiceDate: '03/01/2021',
            dueDate: '03/01/2022',
            invoiceNumber: 'S110220974.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: '31-60',
            customerPo: '109687',
            invoiceDate: '03/01/2021',
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227973.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          }
        ],
        totalAmt: 35232.27,
        totalPastDue: 7.3085938
      }
    }
  }
};

export const successMincron = {
  request: {
    query: InvoicesDocument,
    variables: {
      accountId: '123456',
      erpName: 'MINCRON'
    }
  },
  result: {
    data: {
      invoices: {
        bucketNinety: 0.0,
        bucketOneTwenty: 0.0,
        bucketSixty: 0.0,
        bucketThirty: 0.0,
        bucketFuture: 0.0,
        currentAmt: 35224.96,
        invoices: [
          {
            age: 'Future',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227972.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobNumber: '6566',
            jobName: 'NEP 7 ASHVELLE'
          },
          {
            age: 'Current',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227975.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobNumber: '56787',
            jobName: 'WALTER HORN'
          },
          {
            age: 'Over120',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110220294.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Closed',
            invoiceUrl: 'test.com',
            jobNumber: 'SHOP',
            jobName: 'SEQUAYESH TR'
          },
          {
            age: '91-120',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110222345.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobNumber: 'RETREAT',
            jobName: 'RETREAT AT ARDEN'
          },
          {
            age: 'Deposit',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110220974.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Closed',
            invoiceUrl: 'test.com',
            jobNumber: 'AVLNEP7',
            jobName: 'WALTER HORN'
          },
          {
            age: '31-60',
            customerPo: '109687',
            invoiceDate: '03/01/2021',
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227973.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobNumber: 'AIKENON',
            jobName: 'AIKEN AVL'
          }
        ],
        totalAmt: 35232.27,
        totalPastDue: 7.3085938
      }
    }
  }
};

export const mockTestTableMemo = {
  invoices: {
    bucketNinety: 0.0,
    bucketOneTwenty: 0.0,
    bucketSixty: 0.0,
    bucketThirty: 0.0,
    bucketFuture: 0.0,
    currentAmt: 35224.96,
    invoices: [
      {
        age: 'Future',
        customerPo: '109687',
        invoiceDate: '05/02/2022',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110227972.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: '6566',
        jobName: 'NEP 7 ASHVELLE'
      },
      {
        age: 'Current',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110227975.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: '56787',
        jobName: 'WALTER HORN'
      },
      {
        age: 'Over120',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110220294.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: 'SHOP',
        jobName: 'SEQUAYESH TR'
      },
      {
        age: '91-120',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110222345.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: 'RETREAT',
        jobName: 'RETREAT AT ARDEN'
      },
      {
        age: 'Deposit',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110220974.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: 'AVLNEP7',
        jobName: 'WALTER HORN'
      },
      {
        age: '31-60',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110227973.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: 'AIKENON',
        jobName: 'AIKEN AVL'
      }
    ],
    totalAmt: 35232.27,
    totalPastDue: 7.3085938
  }
};

const rowToMock = success.result.data.invoices.invoices[0];
export const mockInvoiceRow = mockRow<Invoice>(rowToMock, rowToMock, {});
