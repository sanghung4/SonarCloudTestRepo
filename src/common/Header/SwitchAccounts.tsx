import { useContext, useMemo } from 'react';

import {
  Box,
  Divider,
  Grid,
  Link,
  Tooltip,
  Typography,
  useScreenSize
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';
import { Link as RouterLink, useHistory, useLocation } from 'react-router-dom';

import { AuthContext } from 'AuthProvider';

import { switchAccountsStyles } from 'common/Header/util';
import { useUserAccountsQuery } from 'generated/graphql';
import { truncateText } from 'utils/strings';
import { WarningIcon } from 'icons';
import { useSelectedAccountsContext } from 'providers/SelectedAccountsProvider';
import { useCartContext } from 'providers/CartProvider';

/**
 * Styles
 */
const { ChangeButton, tooltipComponentProps } = switchAccountsStyles;

/**
 * Types
 */
type Props = {
  onChange?: () => void;
};

function SwitchAccounts(props: Props) {
  /**
   * Custom hooks
   */
  const history = useHistory();
  const location = useLocation();
  const { isSmallScreen } = useScreenSize();
  const { t } = useTranslation();

  /**
   * Context
   */
  const { authState, profile } = useContext(AuthContext);
  const { contract } = useCartContext();
  const { selectedAccounts } = useSelectedAccountsContext();

  /**
   * Data
   */
  const { data: userAccountsQuery } = useUserAccountsQuery({
    skip: !profile?.userId,
    variables: { userId: profile?.userId },
    onError: (error) => history.push('/error', { error })
  });

  /**
   * Memo
   */
  const userData = useMemo(userDataMemo, [
    contract,
    selectedAccounts,
    userAccountsQuery
  ]);

  /**
   * Callbacks
   */
  const onSwitchPressed = () => {
    history.push('/select-accounts');
    props.onChange?.();
  };

  /**
   * Render
   */
  if (
    !authState?.isAuthenticated ||
    location.pathname === '/select-accounts' ||
    !selectedAccounts.billTo ||
    (!userData.availableShipTos.length &&
      userData.availableBillTos.length === 1)
  ) {
    return null;
  }

  if (isSmallScreen) {
    return (
      <>
        <Box pt={2}>
          <Grid container>
            <Grid item xs={8}>
              <Box pb={1}>
                <Typography color="primary" variant="caption" fontWeight={500}>
                  {t('common.account')}:
                </Typography>
              </Box>
              <Box pb={1} data-testid="mobile-menu-bill-to">
                <Typography noWrap variant="body2">
                  {selectedAccounts.billTo?.name ?? ''}
                </Typography>
              </Box>
              <Box pb={1}>
                <Typography color="primary" variant="caption" fontWeight={500}>
                  {t('common.job')}:
                </Typography>
              </Box>
              <Box pb={3} data-testid="mobile-menu-ship-to">
                <Typography noWrap variant="body2">
                  {userData.shipToName}
                </Typography>
              </Box>
            </Grid>
            <Grid item xs={4}>
              {userData.showButton && (
                <Box pl={7}>
                  <ChangeButton
                    sx={{ fontSize: '0.8rem' }}
                    color="primaryLight"
                    data-testid="change-button"
                    variant="inline"
                    fullWidth={false}
                    onClick={onSwitchPressed}
                  >
                    {t('common.change')}
                  </ChangeButton>
                </Box>
              )}
            </Grid>
          </Grid>
        </Box>
        <Divider />
      </>
    );
  }

  return (
    <Box display="flex" alignItems="center">
      {!!(
        selectedAccounts.billToErpAccount?.creditHold ||
        selectedAccounts.shipToErpAccount?.creditHold
      ) && (
        <Tooltip
          title={t('common.accountHold') as string}
          componentsProps={tooltipComponentProps}
        >
          <Box
            component={WarningIcon}
            mr={1}
            color="secondary.main"
            height={24}
            width={24}
          />
        </Tooltip>
      )}
      {!!userData.shipToName && (
        <Tooltip
          placement="top"
          componentsProps={tooltipComponentProps}
          title={
            <>
              <Typography
                variant="caption"
                component="p"
                fontSize={14}
                lineHeight="22px"
                data-testid="header-tooltip-bill-to"
              >
                {truncateText(selectedAccounts.billTo?.name ?? '', 45)}
              </Typography>
              {selectedAccounts.billTo?.name !== userData.shipToName && (
                <Typography
                  variant="caption"
                  component="p"
                  fontSize={14}
                  lineHeight="22px"
                  data-testid="header-tooltip-ship-to"
                >
                  {truncateText(userData.shipToName, 45)}
                </Typography>
              )}
            </>
          }
        >
          <Typography
            variant="caption"
            fontSize={14}
            lineHeight="22px"
            data-testid="header-ship-to"
          >
            <Box component="span" fontWeight={700}>
              {t('common.job')}
              {': '}
            </Box>
            {truncateText(userData.shipToName, 30)}
          </Typography>
        </Tooltip>
      )}
      {userData.showButton && (
        <Link
          component={RouterLink}
          data-testid="change-link"
          to="/select-accounts"
          underline="none"
          sx={{ p: 0, mr: 0, ml: 1 }}
        >
          <ChangeButton
            data-testid="switch-account-button"
            color="primaryLight"
            variant="text"
            size="small"
          >
            {t('common.switchAccount')}
          </ChangeButton>
        </Link>
      )}
    </Box>
  );

  function userDataMemo() {
    const availableBillTos = userAccountsQuery?.userAccounts ?? [];
    const selectedBillTo = availableBillTos.find(
      (item) => item?.id === selectedAccounts?.billTo?.id
    );
    const availableShipTos = selectedBillTo?.shipTos ?? [];

    const showButton =
      (availableBillTos.length > 1 || availableShipTos.length > 1) && !contract;

    const contractJobName = contract?.data?.jobName;
    const contractAddress =
      contract?.data?.accountInformation?.shipToAddress?.address1;

    const shipToName =
      contractJobName && contractAddress
        ? `${contractJobName} ${contractAddress}`
        : selectedAccounts.shipTo?.name ?? '';

    return {
      availableBillTos,
      availableShipTos,
      showButton,
      shipToName
    };
  }
}

export default SwitchAccounts;
