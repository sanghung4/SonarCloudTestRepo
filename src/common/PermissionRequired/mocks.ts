import { UserContextType } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';

export const authConfigNoPerm = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
export const authConfig = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    userId: 'test',
    permissions: [Permission.EDIT_LIST, Permission.APPROVE_CART],
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
