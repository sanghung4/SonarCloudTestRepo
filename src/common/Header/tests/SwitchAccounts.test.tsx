import { act, fireEvent } from '@testing-library/react';

import { mockCartContext, mockCartContract } from 'Cart/tests/mocks';

import SwitchAccounts from 'common/Header/SwitchAccounts';
import {
  authConfigLoggedIn,
  authConfigLoggedOut
} from 'common/Header/tests/mocks';
import {
  mockUserAccountQueryError,
  mockUserAccountQueryMulti,
  mockUserAccountQueryMultiShipTo,
  mockUserAccountQuerySuccess
} from 'common/Header/tests/SwitchAccounts.mock';
import { CartContext, CartContextType } from 'providers/CartProvider';
import { setBreakpoint } from 'test-utils/mockMediaQuery';
import { render } from 'test-utils/TestWrapper';

const mockHistory = { push: jest.fn() };

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => mockHistory
}));

describe('common - Header - SwitchAccounts', () => {
  afterEach(() => {
    mockHistory.push = jest.fn();
  });
  it('expect to match snapshot on desktop as logged out user', async () => {
    setBreakpoint('desktop');
    const { container } = render(<SwitchAccounts />, {
      authConfig: authConfigLoggedOut,
      selectedAccountsConfig: {
        selectedAccounts: {
          billTo: { id: '1', name: 'bill to' },
          shipTo: { id: '2', name: 'ship to' },
          shippingBranchId: '3'
        }
      }
    });
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(container).toMatchSnapshot();
  });
  it('expect to match snapshot on mobile as logged out user', async () => {
    setBreakpoint('mobile');
    const { container } = render(<SwitchAccounts />, {
      authConfig: authConfigLoggedOut,
      selectedAccountsConfig: {
        selectedAccounts: {
          billTo: { id: '1', name: 'bill to' },
          shipTo: { id: '2', name: 'ship to' },
          shippingBranchId: '3'
        }
      }
    });
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(container).toMatchSnapshot();
  });

  it('expect to match snapshot on desktop as logged in user', async () => {
    setBreakpoint('desktop');
    const { container } = render(<SwitchAccounts />, {
      authConfig: authConfigLoggedIn,
      mocks: [mockUserAccountQuerySuccess],
      selectedAccountsConfig: {
        selectedAccounts: {
          billTo: { id: '1', name: 'bill to' },
          shipTo: { id: '2', name: 'ship to' },
          shippingBranchId: '3'
        }
      }
    });
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(container).toMatchSnapshot();
  });
  it('expect to match snapshot on mobile as logged in user', async () => {
    setBreakpoint('mobile');
    const { container } = render(<SwitchAccounts />, {
      authConfig: authConfigLoggedIn,
      mocks: [mockUserAccountQuerySuccess],
      selectedAccountsConfig: {
        selectedAccounts: {
          billTo: { id: '1', name: 'bill to' },
          shipTo: { id: '2', name: 'ship to' },
          shippingBranchId: '3'
        }
      }
    });
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(container).toMatchSnapshot();
  });

  it('expect to match snapshot on desktop as logged in user with multiple accounts', async () => {
    setBreakpoint('desktop');
    const { container } = render(<SwitchAccounts />, {
      authConfig: authConfigLoggedIn,
      mocks: [mockUserAccountQueryMulti],
      selectedAccountsConfig: {
        selectedAccounts: {
          billTo: { id: '1', name: 'bill to' },
          shipTo: { id: '2', name: 'ship to' },
          shippingBranchId: '3'
        }
      }
    });
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(container).toMatchSnapshot();
  });
  it('expect to match snapshot on mobile as logged in user with multiple accounts', async () => {
    setBreakpoint('mobile');
    const { container } = render(<SwitchAccounts />, {
      authConfig: authConfigLoggedIn,
      mocks: [mockUserAccountQueryMulti],
      selectedAccountsConfig: {
        selectedAccounts: {
          billTo: { id: '1', name: 'bill to' },
          shipTo: { id: '2', name: 'ship to' },
          shippingBranchId: '3'
        }
      }
    });
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(container).toMatchSnapshot();
  });

  it('expect to match snapshot on desktop as logged in user with multiple shipto accounts', async () => {
    setBreakpoint('desktop');
    const { container } = render(<SwitchAccounts />, {
      authConfig: authConfigLoggedIn,
      mocks: [mockUserAccountQueryMultiShipTo],
      selectedAccountsConfig: {
        selectedAccounts: {
          billTo: { id: '1', name: 'bill to' },
          shipTo: { id: '2', name: 'ship to' },
          shippingBranchId: '3'
        }
      }
    });
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(container).toMatchSnapshot();
  });
  it('expect to match snapshot on mobile as logged in user with multiple shipto accounts', async () => {
    setBreakpoint('mobile');
    const { container } = render(<SwitchAccounts />, {
      authConfig: authConfigLoggedIn,
      mocks: [mockUserAccountQueryMultiShipTo],
      selectedAccountsConfig: {
        selectedAccounts: {
          billTo: { id: '1', name: 'bill to' },
          shipTo: { id: '2', name: 'ship to' },
          shippingBranchId: '3'
        }
      }
    });
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(container).toMatchSnapshot();
  });

  it('expect to match snapshot on desktop as logged in user with shipTo accounts on hold', async () => {
    setBreakpoint('desktop');
    const { container } = render(<SwitchAccounts />, {
      authConfig: authConfigLoggedIn,
      mocks: [mockUserAccountQuerySuccess],
      selectedAccountsConfig: {
        selectedAccounts: {
          billTo: { id: '1', name: null },
          shipTo: { id: '2', name: 'ship to' },
          shippingBranchId: '3',
          shipToErpAccount: { creditHold: true }
        }
      }
    });
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(container).toMatchSnapshot();
  });

  it('expect to call history.push on data error', async () => {
    setBreakpoint('desktop');
    render(<SwitchAccounts />, {
      authConfig: authConfigLoggedIn,
      mocks: [mockUserAccountQueryError],
      selectedAccountsConfig: {
        selectedAccounts: {
          billTo: { id: '1', name: 'bill to' },
          shipTo: { id: '2', name: 'ship to' },
          shippingBranchId: '3'
        }
      }
    });
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(mockHistory.push).toBeCalledWith('/error', {
      error: new Error(mockUserAccountQueryError.error!.message)
    });
  });

  it('expect to call onChange and history.push when the Change account is clicked', async () => {
    setBreakpoint('mobile');

    const onChange = jest.fn();
    const { getByTestId } = render(<SwitchAccounts onChange={onChange} />, {
      authConfig: authConfigLoggedIn,
      mocks: [mockUserAccountQueryMulti],
      selectedAccountsConfig: {
        selectedAccounts: {
          billTo: {
            id: 'f8149f07-735b-4b54-81c7-07e39d3749a5',
            name: 'bill to'
          },
          shipTo: {
            id: '2882e752-7323-41cc-976a-7bec199aaa9a',
            name: 'ship to'
          },
          shippingBranchId: '3'
        }
      }
    });
    await act(() => new Promise((res) => setTimeout(res, 0)));
    fireEvent.click(getByTestId('change-button'));
    expect(mockHistory.push).toBeCalledWith('/select-accounts');
    expect(onChange).toBeCalled();
  });

  it('expect to match snapshot with cart contract data', async () => {
    setBreakpoint('desktop');
    const mockCart: CartContextType = {
      ...mockCartContext,
      contract: mockCartContract
    };
    const { container } = render(<SwitchAccounts />, {
      cartConfig: mockCart,
      authConfig: authConfigLoggedIn,
      mocks: [mockUserAccountQuerySuccess],
      selectedAccountsConfig: {
        selectedAccounts: {
          billTo: { id: '1', name: 'bill to' },
          shipTo: { id: '2', name: 'ship to' },
          shippingBranchId: '3'
        }
      }
    });
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(container).toMatchSnapshot();
  });

  it('should link to select accounts', async () => {
    // Arrange
    const { queryByTestId } = render(<SwitchAccounts />, {
      authConfig: { authState: { isAuthenticated: true } },
      selectedAccountsConfig: {
        selectedAccounts: {
          billTo: { id: '1', name: 'bill to' },
          shipTo: { id: '2', name: 'ship to' },
          shippingBranchId: '3'
        }
      }
    });
    const link = queryByTestId('change-link');
    // Assert - link should not be rendered unless there are multiple accounts
    expect(link).toEqual(null);
  });
});
