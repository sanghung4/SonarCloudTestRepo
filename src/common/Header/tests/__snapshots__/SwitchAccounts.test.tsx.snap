// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common - Header - SwitchAccounts expect to match snapshot on desktop as logged in user 1`] = `<div />`;

exports[`common - Header - SwitchAccounts expect to match snapshot on desktop as logged in user with multiple accounts 1`] = `
<div>
  <div
    class="MuiBox-root css-70qvj9"
  >
    <span
      class="MuiTypography-root MuiTypography-caption css-189rpnh-MuiTypography-root"
      data-mui-internal-clone-element="true"
      data-testid="header-ship-to"
    >
      <span
        class="MuiBox-root css-1vg6q84"
      >
        Job
        : 
      </span>
      ship to
    </span>
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-2y5mqq-MuiTypography-root-MuiLink-root"
      data-testid="change-link"
      href="/select-accounts"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-disableElevation MuiButtonBase-root css-mvtbqb-MuiButtonBase-root-MuiButton-root"
        data-testid="switch-account-button"
        tabindex="0"
        type="button"
      >
        Switch Account
      </button>
    </a>
  </div>
</div>
`;

exports[`common - Header - SwitchAccounts expect to match snapshot on desktop as logged in user with multiple shipto accounts 1`] = `<div />`;

exports[`common - Header - SwitchAccounts expect to match snapshot on desktop as logged in user with shipTo accounts on hold 1`] = `<div />`;

exports[`common - Header - SwitchAccounts expect to match snapshot on desktop as logged out user 1`] = `<div />`;

exports[`common - Header - SwitchAccounts expect to match snapshot on mobile as logged in user 1`] = `<div />`;

exports[`common - Header - SwitchAccounts expect to match snapshot on mobile as logged in user with multiple accounts 1`] = `
<div>
  <div
    class="MuiBox-root css-1v3caum"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-15h7jd8-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-4cxybv"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-3wlfnt-MuiTypography-root"
          >
            Account
            :
          </span>
        </div>
        <div
          class="MuiBox-root css-4cxybv"
          data-testid="mobile-menu-bill-to"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap css-15g5syf-MuiTypography-root"
          >
            bill to
          </p>
        </div>
        <div
          class="MuiBox-root css-4cxybv"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-3wlfnt-MuiTypography-root"
          >
            Job
            :
          </span>
        </div>
        <div
          class="MuiBox-root css-hshm0p"
          data-testid="mobile-menu-ship-to"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap css-15g5syf-MuiTypography-root"
          >
            ship to
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1rqx6ho-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-12kl8iw"
        >
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root css-l9fdd-MuiButtonBase-root-MuiButton-root"
            data-testid="change-button"
            tabindex="0"
            type="button"
          >
            Change
          </button>
        </div>
      </div>
    </div>
  </div>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
  />
</div>
`;

exports[`common - Header - SwitchAccounts expect to match snapshot on mobile as logged in user with multiple shipto accounts 1`] = `<div />`;

exports[`common - Header - SwitchAccounts expect to match snapshot on mobile as logged out user 1`] = `<div />`;

exports[`common - Header - SwitchAccounts expect to match snapshot with cart contract data 1`] = `<div />`;
